### DISCUSSION TITLE
I cant upload images to my project.

### DISCUSSION THREAD

**Discussion by albericobenhur** [directus/directus#13002] [Archive]
when i try to upload images to my project i recive the following error:

[SERVICE_UNAVAILABLE] Couldn't save file ddc80e09-8eec-41b6-8957-748d43b6faf2.png

### To Reproduce

go to the archive galery and try to upload some image.

### Errors Shown

On console: Error: Request failed with status code 503
![image](https://user-images.githubusercontent.com/86002589/165162470-775449b6-8fb4-4c1b-88e0-64706e704f6a.png)

### What version of Directus are you using?

9.9.1

### What version of Node.js are you using?

16.14.0

### What database are you using?

Postgres:alpine3.15

### What browser are you using?

firefox

### How are you deploying Directus?

locally

**Comment by rijkvanzanten** [comment]
How are you running Directus? Are you using any third party file storage providers?

**Reply by tenebrius** [reply]
I have the same error. I didnt not configure any storage providers. I just used the docker-compose file provided in the documentation section

**Reply by tenebrius** [reply]
Once i removed the volume entry for the uploads folder, i could upload again. But that also means that the once the docker restarts the contents will be lost

this is the entry i removed
`   # - ./uploads:/directus/uploads`

**Reply by azrikahar** [reply]
>  I didnt not configure any storage providers.

> Once i removed the volume entry for the uploads folder, i could upload again. 

Based on these info, I believe the node user within the docker container does not have the necessary permission to the entry you removed, or rather the host directory you commented out. 

When you removed it, it does have permission since it will definitely be owned by the user _in the container_.

You should be able to either 

- follow the instructions shared here: https://github.com/directus/directus/discussions/6480#discussioncomment-917708

- or do the same thing from the host, via something like `sudo chown -R 1000:1000 ./uploads` (run this on the parent directory, and this assumes node uid/gid is 1000, at least it typically is from my personal experience)



Just to clarify, this is more so something to do with Node.js docker image (as they added a default node user, which may not be what we will face with some other containers in the wild, but is a good practice nonetheless) and Docker in general, rather than Directus itself.

**Reply by tenebrius** [reply]
Thanks for the reply. I was previously running on windows when I encountered the problem. I decided then to run on WSL. At first the got the error `uploads directory is not readable/writeable` though the name volume directory ./uploads was successfully created by docker. I deleted the directory and created manually using simple mkdir, after which it worked perfectly. It seems that even in WSL if the directory is created by docker itself, it does not have the correct permission.
